File: server-mh/src/Match/Handler/Models/PlayerState.ts


export class PlayerState {
    public placeInBoard: number;
    public lights: number;
    public isPresent: boolean;
    public isFinished: boolean;
    public isBot: boolean;
    public hasSpecialSafeCell: boolean;
    public hasSpecialPenaltyCell: boolean;
    public spawnedBefore: boolean;
    

    constructor(
        placeInBoard: number = 0,
        lights: number = 3,
        isPresent: boolean = true,
        isFinished: boolean = false,
        isBot: boolean = false,
        hasSpecialSafeCell: boolean = false,
        hasSpecialPenaltyCell: boolean = false
    ) {
        this.placeInBoard = placeInBoard;
        this.lights = lights;
        this.isPresent = isPresent;
        this.isFinished = isFinished;
        this.isBot = isBot;
        this.hasSpecialSafeCell = hasSpecialSafeCell;
        this.hasSpecialPenaltyCell = hasSpecialPenaltyCell;
        this.spawnedBefore = false;
    }
}

File: Client-unity/Ludo/Assets/Scripts/OfflineMatchHandler/PlayerState.cs
namespace Ludo.Offline
{
    public class PlayerState
    {
        public int PlaceInBoard;
        public int Lights;

        public bool IsFinished;
        public bool IsBot;

        public bool HasSpecialSafeCell;
        public bool HasSpecialPenaltyCell;

        public bool SpawnedBefore;
        public PlayerState()
        {
            PlaceInBoard = -1;
            Lights = 3;

            IsFinished = false;
            IsBot = false;

            HasSpecialSafeCell = true;
            HasSpecialPenaltyCell = true;

            SpawnedBefore = false;
        }
    }
}